streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -502,6 +517,13 @@ public StreamsConfig(final Map<?, ?> props) {
         consumerProps.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, originals().get(BOOTSTRAP_SERVERS_CONFIG));
         // remove deprecate ZK config
         consumerProps.remove(ZOOKEEPER_CONNECT_CONFIG);
+        if (eosEnabled) {
+            if (clientProvidedProps.containsKey(ConsumerConfig.ISOLATION_LEVEL_CONFIG)) {
```

Comment:
We had a discussion about this, and agreed to not throw but overwrite and log. We also created a JIRA for this change: https://issues.apache.org/jira/browse/KAFKA-5096 (with an open PR).

We can discuss again of course. Waiting for further feedback.